File: LeetCodeTasks/11_container-with-most-water/solutions/python3/solution.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        max_water = 0
        while left < right:
            max_water = max(max_water, (right - left) * min(height[left], height[right]))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return max_water

File: LeetCodeTasks/11_container-with-most-water/solutions/c/solution.c
#include <stdio.h>

int maxArea(int* height, int heightSize) {
    int left = 0;
    int right = heightSize - 1;
    int max_water = 0;
    
    while (left < right) {
        int min_height = (height[left] < height[right]) ? height[left] : height[right];
        int current_area = min_height * (right - left);
        if (current_area > max_water) {
            max_water = current_area;
        }
        
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    
    return max_water;
}
